src/data_source/storage/sql.rs

```diff
@@ -1925,6 +1938,539 @@ impl SqlStorage {
+                                ORDER BY (t.block_height, t.index) DESC
+                                LIMIT $3
+                        )
+                        ORDER BY h.height DESC"
+                    ),
+                    vec![
+                        Box::new(block_height as i64),
+                        Box::new(&transaction_index),
+                        Box::new(range.num_transactions.get() as i64),
+                    ],
+                )
+                .await
+            }
+
+            TransactionSummaryFilter::Block(block) => {
+                self.query::<str, Vec<Box<dyn ToSql + Send + Sync>>>(
+                    &format!(
+                        "SELECT {BLOCK_COLUMNS}
+                    FROM header AS h
+                    JOIN payload AS p ON h.height = p.height
+                    WHERE h.height IN (
+                        SELECT t.block_height
+                            FROM transaction AS t
+                            WHERE (t.block_height, t.index) <= ($1, $2)
+                            ORDER BY (t.block_height, t.index) DESC
+                            LIMIT $3
+                    )
+                    AND h.height = $4
+                    ORDER BY h.height DESC"
+                    ),
+                    vec![
+                        Box::new(block_height as i64),
+                        Box::new(&transaction_index),
+                        Box::new(range.num_transactions.get() as i64),
+                        Box::new(*block as i64),
+                    ],
+                )
+                .await
+            }
+        }?
+        .map(|result| match result {
+            Ok(row) => parse_block::<Types>(row),
+            Err(err) => Err(err),
+        });
+
+        let transaction_summary_stream = block_stream.flat_map(|row| match row {
+            Ok(block) => stream::iter(
+                block
+                    .enumerate()
+                    .enumerate()
+                    .map(|(index, (_, txn))| {
+                        TransactionSummary::try_from((&block, index, txn)).map_err(|err| {
+                            QueryError::Error {
+                                message: err.to_string(),
+                            }
+                        })
+                    })
+                    .collect::<Vec<QueryResult<TransactionSummary<Types>>>>()
+                    .into_iter()
+                    .rev()
+                    .collect::<Vec<QueryResult<TransactionSummary<Types>>>>(),
+            ),
+            Err(err) => stream::iter(vec![Err(err)]),
+        });
+
+        let transaction_summary_vec = transaction_summary_stream
+            .try_collect::<Vec<TransactionSummary<Types>>>()
+            .await?;
+
+        Ok(transaction_summary_vec
+            .into_iter()
+            .skip_while(|txn| {
+                if let TransactionIdentifier::Hash(hash) = target {
```

Comment:
why do we need this check? Can we make target included as well in transaction summaries?

Reply:
I'm not entirely following what you're asking here about including the target in the `transaction summaries`.

I'm fairly certain we do need a check for the `Hash` here, as we retrieve `Block`s from the `query`, and we don't know where in the block that the `Transaction` identified by the `Hash` in question occurs.